player.cpp

```diff
@@ -49,22 +51,55 @@ Player Player::parseUnit(const std::string &fileName)
         }
     }
 
-    return Player(properties["name"], stoi(properties["hp"]), stoi(properties["dmg"]));
+    return Player(properties["name"], stoi(properties["hp"]), stoi(properties["dmg"]), 0, 1);
 }
 
-bool Player::Attack(Player *otherPlayer) const
+bool Player::Attack(Player *otherPlayer)
 {
+    bool fatality;
+
     if (otherPlayer->hp <= this->damage)
     {
         otherPlayer->hp = 0;
-        return true;
+        fatality = true;
+    }
+    else
+    {
+        otherPlayer->hp -= this->damage;
+        fatality = false;
     }
 
-    otherPlayer->hp -= this->damage;
-    return false;
+    // Increase XP
+    this->increaseXP(this->damage);
+
+    return fatality;
+}
+
+void Player::increaseXP(unsigned short amount)
+{
+    this->xp += amount;
+    unsigned short properLevel = (this->xp / LEVEL_SIZE) + 1;
+
+    if (properLevel > this->level)
+    {
+        this->levelUp(properLevel);
+    }
```

Comment:
Egy ütéstől akár több szintet is léphet. (Amúgy ha az xp-t így kumulálva tároljátok és nem az utolsó szintlépés óta szerzett xp-t, akkor fölösleges külön tárolni a szintet, mert a fenti képlettel bármikor kiszámítható az xp-ből.)